common/src/main/java/com/microsoft/identity/common/internal/ui/webview/challengehandlers/SwitchBrowserUriHelper.kt

```diff
@@ -0,0 +1,117 @@
+     * e.g. msauth://com.microsoft.identity.client/your-redirect-uri?code=your-switch-browser-code&action_uri=your-action-uri
+     *
+     * @return The process uri constructed from the broker redirect uri.
+     * e.g. your-action-uri?code=your-switch-browser-code
+     */
+    fun buildProcessUri(brokerRedirectUri: Uri): Uri? {
+        val methodTag = "$TAG:buildProcessUri"
+        // Get the SwitchBrowser purpose token from the broker redirect uri.
+        val code = brokerRedirectUri.getQueryParameter(
+            SWITCH_BROWSER.CODE
+        )
+        if (code.isNullOrEmpty()) {
+            // This should never happen, but if it does, we should log it and return.
+            Logger.warn(methodTag, "Switch browser code is null or empty ")
+            return null
+        }
+        // Get the process uri from the broker redirect uri.
+        val actionUri = brokerRedirectUri.getQueryParameter(
+            SWITCH_BROWSER.ACTION_URI
+        )
+        if (actionUri.isNullOrEmpty()) {
+            // This should never happen, but if it does, we should log it and return.
+            Logger.warn(methodTag, "Switch browser action URI is null or empty ")
+            return null
+        }
+        // Query parameters for the process uri.
+        val queryParams = hashMapOf<String, String>()
+        queryParams[SWITCH_BROWSER.CODE] = code
+        queryParams[OAuth2.REDIRECT_URI] = Broker.NEW_BROKER_REDIRECT_URI
+        // Construct the uri to the process endpoint.
+        return buildSwitchBrowserUri(actionUri, queryParams)
+    }
+
+    /**
+     * Build a generic switch browser uri.
+     *
+     * @param actionUri The action uri to be opened.
+     * @param queryParams The query parameters to be included in the switch browser uri.
+     *
+     * @return The switch browser uri constructed from the action uri and query parameters.
+     */
+    private fun buildSwitchBrowserUri(
+        actionUri: String,
+        queryParams: HashMap<String, String>
+    ): Uri? {
+        val paths = actionUri.split("/")
+        val authority = paths[0]
+        val uriBuilder = Uri.Builder()
+            .scheme("https")
+            .encodedAuthority(authority)
+        for (i in 1 until paths.size) {
+            uriBuilder.appendPath(paths[i])
+        }
+        for ((key, value) in queryParams.entries) {
+            uriBuilder.appendQueryParameter(key, value)
+        }
+        return uriBuilder.build()
+    }
+
+    /**
+     * Check if the request is to switch the browser.
+     *
+     * The request is considered "switch_browser" if the URL
+     * starts with the following pattern: msauth:// Microsoft.AAD.BrokerPlugin/switch_browser
+     *
+     *
+     * @param url The URL to be checked.
+     * @return True if the request contains the required parameters, false otherwise.
+     */
+    fun isSwitchBrowserRequest(url: String?): Boolean {
```

Comment:
I think we can add also a `redirectUri` param here and then use that for matching at line 115. When you call the method then you pass the BROKER_REDIRECT_URI.

This way if we ever need to use with other redirect uri then code is extendable

Reply:
done